src/ConsoleBuffer/Commands/ControlSequence.cs

```diff
@@ -2,9 +2,26 @@ namespace ConsoleBuffer.Commands
 {
     using System;
     using System.Collections.Generic;
+    using System.Diagnostics;
 
     public abstract class ControlSequence : Base
     {
+        public class ParsedParameters
+        {
+            public int GetValue(int offset, int defaultValue = 0, int maxValue = int.MaxValue)
+            {
+                if(offset < this.Count)
+                {
+                    return Math.Max(defaultValue, Math.Min(this.Parameters[offset], maxValue));
+                }
+
+                return defaultValue;
+            }
+
+            public int Count { get; set; }
+            public int[] Parameters { get; set; } = new int[16];
```

Comment:
It's not clear where the magic '16' comes from here? Can this be extended? What happens if a potentially malicious actor sends >16 split params?

Reply:
No magic, I bumped it up to 32, which I hope is enough, and an exception is thrown if the parameter list grows beyond 32.  I'm going to assume something has gone horribly wrong at that point anyway and it should be stopped.

Reply:
What we should probably do here (32 is more than enough!) is not throw an exception, but rather silently swallow additional parameters and somehow mark the sequence as invalid. Exceptions are fairly costly and there's already code in the parser to deal with invalid/dead sequences.